egui_demo_lib/src/apps/demo/paint_bezier.rs

```diff
@@ -0,0 +1,182 @@
+    fn build_auxiliary_line(&self,points:&[Pos2],to_screen:&RectTransform,aux_stroke:&Stroke)->Vec<Shape>{
+        let mut shapes = Vec::new();
+        if points.len()>=2 {
+            let points:Vec<Pos2> = points.iter().map(|p| to_screen * *p).collect();
+            shapes.push(egui::Shape::line(points, aux_stroke.clone()));
+        }
+        for point in points.iter(){
+            let center = to_screen * *point;
+            let radius = aux_stroke.width * 3.0;
+            let circle = CircleShape {
+                center,
+                radius,
+                fill: aux_stroke.color,
+                stroke: aux_stroke.clone(),
+            };
+            
+            shapes.push(circle.into());
+        }
+        
+        shapes
+    }
+
+    pub fn ui_content(&mut self, ui: &mut Ui) -> egui::Response {
+        let (mut response, painter) =
+            ui.allocate_painter(ui.available_size_before_wrap(), Sense::click());
+
+        let to_screen = emath::RectTransform::from_to(
+            Rect::from_min_size(Pos2::ZERO, response.rect.square_proportions()),
+            response.rect,
+        );
+        let from_screen = to_screen.inverse();
+
+        if response.clicked() {
+            if let Some(pointer_pos) = response.interact_pointer_pos() {
+                let canvas_pos = from_screen * pointer_pos;
+                self.points.push(canvas_pos);
+                if self.points.len() >= self.bezier {
+                    self.backup_points = self.points.clone();
+                    let points = self.points.drain(..).collect::<Vec<_>>();
+                    match points.len(){
+                        3 => {
+                            let points: Vec<Pos2> = points.iter().map(|p| to_screen * *p).collect();
+                            let stroke = self.stroke.clone();
+                            let quadratic = QuadraticBezierShape::from_points_stroke(points, stroke);
+                            self.quadratic_shapes.push(quadratic);
+                        },
+                        4 => {
+                            let points: Vec<Pos2> = points.iter().map(|p| to_screen * *p).collect();
+                            let stroke = self.stroke.clone();
+                            let cubic = CubicBezierShape::from_points_stroke(points, stroke);
+                            self.cubic_shapes.push(cubic);
+                        },
+                        _ => {
+                            todo!();
+                        }
+                    }
+                    
+                    // self.points.clear();
+                }
+                
+                response.mark_changed();
+            } 
+        }
+        let mut shapes = Vec::new();
+        for shape in self.quadratic_shapes.iter() {
+            shapes.push(shape.clone().into());
+        }
+        for shape in self.cubic_shapes.iter() {
+            shapes.push(shape.clone().into());
+        }
+        painter.extend(shapes);
```

Comment:
There is a problem with the demo: when you move the window, the shapes stay put.

You need to store the painted shapes in the local coordinate system and then translate them here using `Shape::translate`.

Reply:
I found it's inconvenient to use `translate` here since you have to remember the previous screen window in order to get the delta. Instead, I applied the `to_screen` transform to every point.